src/Package/Impl/ProjectSystem/RDebugLaunchProvider.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.Common.Core;
+using Microsoft.R.Components.InteractiveWorkflow;
+using Microsoft.R.Debugger;
+using Microsoft.R.Debugger.PortSupplier;
+using Microsoft.R.Host.Client;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.Debuggers;
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+using Microsoft.VisualStudio.ProjectSystem.Utilities.DebuggerProviders;
+using Microsoft.VisualStudio.ProjectSystem.VS.Debuggers;
+
+namespace Microsoft.VisualStudio.R.Package.ProjectSystem {
+    // ExportDebugger must match rule name in ..\Rules\Debugger.xaml.
+    [ExportDebugger("RDebugger")] 
+    [AppliesTo(Constants.RtvsProjectCapability)]
+    internal class RDebugLaunchProvider : DebugLaunchProviderBase {
+        private readonly ProjectProperties _properties;
+        private readonly IRInteractiveWorkflow _interactiveWorkflow;
+
+        [ImportingConstructor]
+        public RDebugLaunchProvider(ConfiguredProject configuredProject, IRInteractiveWorkflowProvider interactiveWorkflowProvider)
+            : base(configuredProject) {
+            _properties = configuredProject.Services.ExportProvider.GetExportedValue<ProjectProperties>();
+            _interactiveWorkflow = interactiveWorkflowProvider.GetOrCreate();
+        }
+
+        private IRSession Session => _interactiveWorkflow.RSession;
+
+        public override async Task<bool> CanLaunchAsync(DebugLaunchOptions launchOptions) {
```

Comment:
Should we allow F5 and pop up message telling user to specify function to call in the project properties? It may be hard for them to discover how to make F5 working